### ISSUE TITLE
Laggy and low fps while window is open after scan

### ISSUE THREAD

**Issue by cars11** [windirstat/windirstat#151]
if I leave windirstat open after a scan it reduces my 120hz/fps down to about 10. everything is extremely laggy.
immediately gone after closing windirstat, and does not happen while it is open and the "select drive' window is still up.

**Comment by NoMoreFood** [comment]
@cars11  Can you see if you have those same problems with the beta version? 

https://github.com/windirstat/windirstat/releases/tag/beta/v2.1.2/2024-12-12

**Comment by cars11** [comment]
yes, issue persists

**Comment by NoMoreFood** [comment]
What sort of hardware are you running and what are you using to measure FPS?

**Comment by cars11** [comment]
intel 9900k and an nvidia 3090.

my eyes.

**Comment by NoMoreFood** [comment]
Yeah, that's about what I have and don't notice any slowdown to that extent.  Anything unique about your setup that you think could be consequential?

**Comment by pgwipeout** [comment]
I came here specifically to second this. While the scan was running, even minimized the entire display becomes jittery. Most of it has to do with mouse input skipping, because video only gets affected when it gets particularly bad. It gets particularly bad when the scan is complete and the treemap is displayed. Setting the process to low priority lessens the effect, but it doesn't stop it entirely.

Hardware:
Ryzen 7 5800, sitting at ~25% utilization, no core is maxed out.
64GB ram (Windirstat states it's consuming about 900mb, task manager agrees)
Nvidia 4080
Dual screen, HDR, 120hz monitors, 1.25% dpi scaling, 1440p

**Comment by NoMoreFood** [comment]
@pgwipeout, @cars11  Just to confirm... are you both talking about during or after the scan.  I think during the scan, I would completely expect since there is a lot of math (you can turn down the threads to 1 to lessen the effect), as you mentioned, decrease the priority.  After the scan, there shouldn't be anything noticeable for a modern processor.

**Comment by NoMoreFood** [comment]
@cars11  Thanks for confirming.  I thought that was what you were saying.

**Comment by pgwipeout** [comment]
I experience the jitter any time after a scan starts. It's not the background processes doing it, something with the UI updates is wreaking havoc with the display manager. Lowering the threads only helped because it reduced the number of UI updates that occurred. It is particularly bad when the progress indicators move, and then becomes a terrible mess when the scan completes and the tree is visible. 

It's strange too that minimizing it doesn't stop the jitter, because that should stop UI updates.

**Comment by cars11** [comment]
it didn't happen in the old version 1.xx

**Comment by pgwipeout** [comment]
Precisely, version 1.x was blurry on HiDPI displays though which is why I went to upgrade.

I've found another interesting tidbit, opening Sysinternals Process Explorer (I was going to test suspending and resuming it) fixes the jitter, except when the mouse cursor moves over the process list. Minimizing the Process Explorer window causes the jitter to resume. Suspending the process doesn't stop the jitter entirely, so this is an issue between the process and WDM. 

I'm going to try messing with the HiDPI, HDR, and VRR settings to see if I can isolate this down further.

**Comment by pgwipeout** [comment]
It seem the only thing that affects it is the amount of data processed. Once the memory usage gets above ~600mb the whole computer starts slowing down. I disabled the animated pacman guys and it helped with the jitter, but otherwise there's no hope for it.

**Comment by pgwipeout** [comment]
It's funny, toggling HDR off and on again stopped the jitter as well, but didn't help the slow down. Does the UI run on a separate thread than the scan and the data management?

**Comment by NoMoreFood** [comment]
I didn't do anything to explicitly address this (since I can't reproduce the behavior), but I made several performance tweaks in the latest beta if you care to check it out: https://github.com/windirstat/windirstat/releases/tag/beta/v2.1.2/2024-12-17

**Comment by pgwipeout** [comment]
Running a scan now, I can say previously I was sitting about 30% CPU usage total during a scan, now it's a race between the antivirus watching the scans and WinDirStat scanning the files for who can peg the CPU completely. So yeah, speed is significantly improved.

**Comment by pgwipeout** [comment]
I can say with absolute certainty that the 18 Dec beta corrected my issue. It also flies!
Good job fixing the memory leak from the 17 Dec beta too.

**Comment by pgwipeout** [comment]
@NoMoreFood Interestingly enough, after staying open for 24+ hours with a complete scan, it started to get a little laggy on the 18 Dec beta. Also, while force deleting duplicate files, it crashed with the following:
` Unhandled exception at 0x00007FF6EBEF757C in WinDirStat.exe: 0xC000041D: An unhandled exception was encountered during a user callback.`

**Comment by NoMoreFood** [comment]
@pgwipeout I'll try leaving it open as well as see what happens.  I cannot imagine why behavior should change over time.  I did fix an issue or two related to deleting items in release I just made.

**Comment by an4120** [comment]
I am still getting this issue even on the latest beta. All I have done is ran a scan and left tthe app open afterwards. Win explorer feels laggy and takes six seconds to open a video file. It all goes away when I close the app.

**Comment by NoMoreFood** [comment]
@an4120 That's crazy.  The only thing that happens after the scan (that's not prompted by user interactions) is the memory usage information is updated every half second. Does the behavior change if you minimize it?  If it's minimized, a timer still fires every 25ms but the function it calls exits it immediately.

**Comment by an4120** [comment]
> @an4120 That's crazy. The only thing that happens after the scan (that's not prompted by user interactions) is the memory usage information is updated every half second. Does the behavior change if you minimize it? If it's minimized, a timer still fires every 25ms but the function it calls exits it immediately.

The choppy laggy feeling of my computer about to melt seems to be gone with the 12-28 version, however opening a video file still takes like 4 seconds.  It takes the longest, to an annoying extent, when the app is open on the desktop; it goes faster when it is minimized, and fastest when it is closed.  To clarify, i'm opening files with mpc-hc 2.3.5. It seems like vlc is unaffected by all this and just instantly opens the videos. 
If it matters, I scanned my media server with windirstat which is 126.6 TB in size and was using 176 MB "Memory Usage" on the app. 
It also akes awhile for the problem to appear - it initially works fine. I'm guessing it took probably 10-15mins to notice it happening.

**Comment by NoMoreFood** [comment]
@an4120 Thanks for the update.  I did find one thing of interest where my timer loop might have been triggering some expensive built-in control update logic so the next release will update that.

**Comment by cars11** [comment]
Yeah, the fact that video files and VLC are unusable makes it particularly annoying

**Comment by NoMoreFood** [comment]
@cars11    Can you see if this is any better for you?

https://github.com/windirstat/windirstat/releases/tag/beta%2Fv2.1.2%2F2024-12-29

**Comment by an4120** [comment]
> @cars11 @an4120 Can you see if this is any better for you?
> 
> https://github.com/windirstat/windirstat/releases/tag/beta%2Fv2.1.2%2F2024-12-29

Same as before. I scanned my media server, left the window open, and videos take like 4 seconds to load when launching them with the latest version of MPC-HC. During this time when the video is "launching / loading" fps drops down low causing the experience to be very choppy. 
The videos load much quciker when windirstat is minimized. 

I do not experience this problem if I Launch/Load the video by opening it with VLC media player. 

Maybe I could make a recording of what I am  experiencing for you ?

**Comment by NoMoreFood** [comment]
@an4120 Also, another release to try.  I found a small handle leak that I addressed.

https://github.com/windirstat/windirstat/releases/tag/beta%2Fv2.1.2%2F2024-12-30

**Comment by an4120** [comment]
> @an4120 @cars11 Also, another release to try. I found a small handle leak that I addressed.
> 
> https://github.com/windirstat/windirstat/releases/tag/beta%2Fv2.1.2%2F2024-12-30

I'll check that out. Here are 2 videos documenting what I am experiencing. You can see the FPS fluctuating on the bottom right corner of the OBS recording window. 
https://gofile.io/d/uTENtv
First video was recording with nvenc, then I switched to cpu x264 for the second one.

**Comment by an4120** [comment]
> @an4120 @cars11 Also, another release to try. I found a small handle leak that I addressed.
> 
> https://github.com/windirstat/windirstat/releases/tag/beta%2Fv2.1.2%2F2024-12-30

I think this may have fixed it for me. 

I ran into a different error where I turned my server off (Which had been scanned) before shutting the program down and it seemed to lock windirstat up. 
Now the program takes like 20-30 seconds to start up on the "select drives" menu . :-(

**Comment by NoMoreFood** [comment]
@an4120 Well good to know we're making _some_ progress.  Are you able to consistently reproduce the new behavior you are describing?

**Comment by pgwipeout** [comment]
The 12-30 beta is even better, stutter later on during large scans is gone, but memory usage has doubled (5.5GB for all local drives). I'll leave it open for a while to see if the slowdown issue returns.

**Comment by pgwipeout** [comment]
It would appear the issue causing the computer to lag out is gone, now the only thing that gets laggy is the program itself.

**Comment by NoMoreFood** [comment]
@pgwipeout Can you see if memory consumption has improved on the latest beta:

https://github.com/windirstat/windirstat/releases/tag/beta/v2.1.2/2025-01-02

**Comment by NoMoreFood** [comment]
Released with https://github.com/windirstat/windirstat/releases/tag/release/v2.2.0

If any additional action is warranted, please submit a new issue.
